var/spack/repos/builtin/packages/sysbench/package.py

```diff
@@ -17,11 +17,40 @@ class Sysbench(AutotoolsPackage):
     version("1.0.19", sha256="39cde56b58754d97b2fe6a1688ffc0e888d80c262cf66daee19acfb2997f9bdd")
     version("1.0.18", sha256="c679b285e633c819d637bdafaeacc1bec13f37da5b3357c7e17d97a71bf28cb1")
 
+    variant("mysql", default=False, description="Compile with MySQL support")
+    variant("pgsql", default=False, description="Compile with PostgreSQL support")
+
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
 
     depends_on("autoconf", type="build")
     depends_on("automake", type="build")
+    depends_on("binutils+ld+gold", type="build")  # required when system doesn't provide zlib
     depends_on("libtool", type="build")
     depends_on("m4", type="build")
-    depends_on("mysql-client")
+    depends_on("pkgconfig", type="build")
+    depends_on("mysql-client", when="+mysql")
+    depends_on("postgresql", when="+pgsql")
+    depends_on("zlib-api", type="build")
+
+    phases = ["autoreconf", "configure", "build", "install"]
+
+    @property
+    def zlib_library_path(self):
+        return self.spec["zlib-api"].prefix.lib
+
+    def flag_handler(self, name, flags):
+        if name == "ldflags":
+            flags.append(f"-L{self.zlib_library_path}")
+        return flags, None, None
+
+    def configure_args(self):
+        args = []
+
+        args += self.with_or_without("mysql")
+        args += self.with_or_without("pgsql")
+
+        return args
+
+    def autoreconf(self, spec, prefix):
+        which("bash")("./autogen.sh")
```

Comment:
`autogen.sh` is a `/bin/sh` script, so we can use:
```suggestion
        Executable("./autogen.sh")()
```
(otherwise we must depend on `bash`)